### PR TITLE
Added mod function for complex numbers

### PR DESCRIPTION
**Pull request by prithvitewatia**
This pull request addresses the creation of a new mod function as discussed in issue #37376.
Note: @stevengj Please review this pull request. I apologize for closing the previous pull request as I was not able to sync with main branch properly and messed up the commits. I will ensure that such mistakes will not be repeated in future.

### CODE REVIEW

**Comment by prithvitewatia** [comment]
Also could you please guide me on why this buildbot/tester_win64 is failing because upon my inspection this error is coming from a file that I did not change and my branch is synchronized with the Julia master branch.

File: base/complex.jl [hunk 1]
```diff
@@ -263,6 +263,20 @@ julia> conj(1 + 3im)
...
+Returns a complex number whose real part and imaginary parts are old real part modulo n and old imaginary part modulo n respectively.
+# Examples
+```jldoctest
+julia> mod(12+10im,5)
+2 + 0im
+julia> mod(3+11im,7)
+3 + 4im
+```
+"""
+mod(z::Complex,n::Real)=Complex(mod(real(z),n),mod(imag(z),n))
```
Comment (stevengj) [inline base/complex.jl:278]: It's not clear that we need a new docstring, since the old docstring for `mod(x, y)` seems to cover this case.

However, for consistency we should have similar methods for `div` and `rem`

File: NEWS.md [hunk 2]
```diff
@@ -33,7 +33,9 @@ New library functions
 * Two argument methods `findmax(f, domain)`, `argmax(f, domain)` and the corresponding `min` versions ([#27613]).
 * `isunordered(x)` returns true if `x` is value that is normally unordered, such as `NaN` or `missing`.
 * New macro `Base.@invokelatest f(args...; kwargs...)` provides a convenient way to call `Base.invokelatest(f, args...; kwargs...)` ([#37971])
-* New macro `Base.@invoke f(arg1::T1, arg2::T2; kwargs...)` provides an easier syntax to call `invoke(f, Tuple{T1,T2}, arg1, arg2; kwargs...)` ([#38438])
+* New macro `Base.@invoke f(arg1::T1, arg2::T2; kwargs...)` provides an easier syntax to call `invoke(f, Tuple{T1,T2}; kwargs...)` ([#38438])
```
Comment (stevengj) [inline NEWS.md:36]: This line shouldn't have been changed.

File: base/refpointer.jl [hunk 3]
```diff
@@ -137,6 +137,8 @@ if is_primary_base_module
 
     Ref(x::Ref, i::Integer) = (i != 1 && error("Ref only has one element"); x)
     Ref(x::Ptr{T}, i::Integer) where {T} = x + (i - 1) * Core.sizeof(T)
+    #For multidimensional-arrays
+    Ref(x::AbstractArray{T},row::Integer,column::Integer) where {T} = Ref(x,row+(column-1)*size(x)[1])
```
Comment (stevengj) [inline base/refpointer.jl:141]: This should also not have been changed — did you accidentally pull in an unrelated change?

File: test/complex.jl [hunk 4]
```diff
@@ -1188,6 +1188,36 @@ end
     @test tanh(atanh(complex(-1.0,-1.0))) == complex(-1.0,-1.0)
 end
 
+@testset "mod, issue #37376" begin
+    #Testing on zeros and ones
+    @test isequal(mod(0+0im,1),Complex(0,0))
+    @test isequal(mod(Complex(0,0),2),Complex(0,0))
+    @test isequal(mod(Complex(1,0),1),Complex(0,0))
+    @test isequal(mod(Complex(0,1),1),Complex(0,0))
+    @test isequal(mod(Complex(1,1),1),Complex(0,0))
```
Comment (stevengj) [inline test/complex.jl:1197]: For this kind of repeated code, you can do:
```
for (z,n) in ((0+0im,1), (0+0im,2), (1+0im,1), (0+1im), 1), (1+1im,1))
    @test mod(z, n) === 0+0im
end
```

File: test/complex.jl [hunk 5]
```diff
@@ -1188,6 +1188,36 @@ end
...
+    #Testing on integer numerator and denominator
+    @test isequal(mod(Complex(5,6),2),Complex(1,0))
+    @test isequal(mod(Complex(11,17),7),Complex(4,3))
+    @test isequal(mod(Complex(10,12),5),Complex(0,2))
```
Comment (stevengj) [inline test/complex.jl:1201]: Probably it's sufficient to just check one case for each type, since the underlying `mod` function is tested elsewhere — we just want to verify that complex numbers pass through to the real `mod`.

File: NEWS.md (hunk 2 above)
Comment (prithvitewatia) [reply NEWS.md:36]: Sir The new line was causing a conflict with my branch so I just kept the original JuliaLang/Julia News.md file.

File: base/refpointer.jl (hunk 3 above)
Comment (prithvitewatia) [reply base/refpointer.jl]: Sir, I was working on a different issue ( #39672 ). This by mistake got pulled in this branch. I have removed this code.

**Comment by prithvitewatia** [comment]
@stevengj friendly reminder. Sir, you haven't reviewed my pull request since the last changes. Please take a look.

File: NEWS.md (hunk 2 above)
Comment (stevengj) [reply NEWS.md:36]: You may need to do a rebase to fix the conflict?

Comment (prithvitewatia) [reply NEWS.md:36]: Sir, but now the conflict is already fixed and I have also added the new functions in the respective files.
